### PR TITLE
Add webcrypto aes-gcm opitonal aad

### PR DESCRIPTION
**Pull request by kigawas**
- Add AAD
- Separate aes-gcm and aes-gcm-siv tests

### CODE REVIEW

File: test/gcm-siv.test.js [hunk 1]
```diff
@@ -0,0 +1,893 @@
...
+    polyvalInput:
+      'eb3640277c7ffd1303c7a542d02d3e4c000000000000000000000000000000000000000000000000c000000000000000',
+    polyvalResult: '7367cdb411b730128dd56e8edc0eff56',
+    polyvalResultXOR: '7367cdb411b730128dd56e8edc0eff56',
+    polyvalResultMasked: '7367cdb411b730128dd56e8edc0eff56',
+    tag: 'ffffffff000000000000000000000000',
+    counter: 'ffffffff000000000000000000000080',
+    result: '18ce4f0b8cb4d0cac65fea8f79257b20888e53e72299e56dffffffff000000000000000000000000',
+  },
+];
```
Comment (paulmillr) [inline test/gcm-siv.test.js:829]: Could you place SIV vectors into `test/vectors/siv.json`?

File: src/webcrypto/crypto.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 // We use WebCrypto aka globalThis.crypto, which exists in browsers and node.js 16+.
 // See utils.ts for details.
 declare const globalThis: Record<string, any> | undefined;
-export const crypto =
+export const crypto: Crypto =
```
Comment (paulmillr) [inline src/webcrypto/crypto.ts:4]: Remove this, it adds dependency on `dom`

File: src/webcrypto/aes.ts [hunk 3]
```diff
@@ -1,33 +1,59 @@
...
 import { getWebcryptoSubtle } from './utils.js';
 
-function generate(algo: string, length: number) {
+/**
+ * AAD is only effective on AES-256-GCM or AES-128-GCM. Otherwise it'll be ignored
+ */
+export type Cipher = (
+  key: Uint8Array,
+  nonce: Uint8Array,
+  AAD?: Uint8Array
```
Comment (paulmillr) [inline src/webcrypto/aes.ts:10]: I wonder if AAD should be here or in encrypt/decrypt?

**Review by paulmillr — CHANGES_REQUESTED** [review]

File: src/webcrypto/aes.ts (hunk 3 above)
Comment (kigawas) [reply src/webcrypto/aes.ts:10]: `aes_256_gcm_siv` is here. But it's more accurate in `encrypt/decrypt`

File: src/webcrypto/crypto.ts (hunk 2 above)
Comment (kigawas) [reply src/webcrypto/crypto.ts:4]: Is it also necessary to remove it in cryptoNode.js?

File: src/webcrypto/aes.ts (hunk 3 above)
Comment (kigawas) [reply src/webcrypto/aes.ts:10]: I tried to move it to `encrypt/decrypt` but some tests failed

File: src/webcrypto/crypto.ts (hunk 2 above)
Comment (paulmillr) [reply src/webcrypto/crypto.ts]: yes

Comment (kigawas) [reply src/webcrypto/crypto.ts]: Can you review again?

File: test/gcm-siv.test.js (hunk 1 above)
Comment (paulmillr) [reply test/gcm-siv.test.js]: sure, later.

**Pull request merged**